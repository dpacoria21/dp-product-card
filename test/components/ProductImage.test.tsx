import React from 'react';
import rendered from 'react-test-renderer';
import { ProductImage } from '../../src/components/ProductImage';
import ProductCard from '../../src/components/ProductCard';
import { product2 } from '../data/product';



describe('<ProductImage />', () => {
    test('La imagen deberia de aparecer en el componente', () => {

        const wrapper = rendered.create(
            <ProductImage img='saludos.png' />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });


    test('La imagen debe de aparecer al mandar como un product', () => {
        const wrapper = rendered.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})