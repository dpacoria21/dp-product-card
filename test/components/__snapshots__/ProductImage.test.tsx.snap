// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImage /> La imagen debe de aparecer al mandar como un product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`<ProductImage /> La imagen deberia de aparecer en el componente 1`] = `
<img
  alt="Product image"
  className="productImg undefined"
  src="saludos.png"
/>
`;
